packages/eas-cli/src/commands/build/download.ts

```diff
@@ -0,0 +1,138 @@
+
+  static override contextDefinition = {
+    ...this.ContextOptions.LoggedIn,
+    ...this.ContextOptions.ProjectId,
+  };
+
+  async runAsync(): Promise<void> {
+    const {
+      flags: { json: jsonFlag, 'non-interactive': nonInteractive, platform, fingerprint },
+    } = await this.parse(Download);
+
+    const {
+      loggedIn: { graphqlClient },
+      projectId,
+    } = await this.getContextAsync(Download, {
+      nonInteractive,
+    });
+
+    if (jsonFlag) {
+      enableJsonOutput();
+    }
+
+    const selectedPlatform = await resolvePlatformAsync(nonInteractive, platform);
+    const build = await this.getBuildAsync(graphqlClient, projectId, selectedPlatform, fingerprint);
+    const buildPath = await this.getPathToBuildAppAsync(build, selectedPlatform);
+    if (jsonFlag) {
+      const jsonResults: { path?: string } = {};
+      if (buildPath) {
+        jsonResults.path = buildPath;
+      }
+      printJsonOnlyOutput(jsonResults);
+    } else {
+      Log.log(`Build downloaded at ${chalk.bold(buildPath)}`);
+    }
+  }
+
+  private async getBuildAsync(
+    graphqlClient: ExpoGraphqlClient,
+    projectId: string,
+    platform: AppPlatform,
+    fingerprintHash: string
+  ): Promise<BuildFragment> {
+    const builds = await BuildQuery.viewBuildsOnAppAsync(graphqlClient, {
+      appId: projectId,
+      filter: {
+        platform,
+        fingerprintHash,
+        status: BuildStatus.Finished,
+        simulator: platform === AppPlatform.Ios ? true : undefined,
+        distribution: platform === AppPlatform.Android ? DistributionType.Internal : undefined,
+        developmentClient: true,
+      },
+      offset: 0,
+      limit: 1,
+    });
+    if (builds.length === 0) {
+      throw new Errors.CLIError(
+        `No builds available for ${platform} with fingerprint hash ${fingerprintHash}`
+      );
+    }
+
+    Log.succeed(`🎯 Found successful build with matching fingerprint on EAS servers.`);
+    return builds[0];
+  }
+
+  async getPathToBuildAppAsync(build: BuildFragment, platform: AppPlatform): Promise<string> {
+    const cachedAppPath = getEasBuildRunCachedAppPath(build.project.id, build.id, platform);
+    if (await pathExists(cachedAppPath)) {
+      Log.newLine();
+      Log.log(`Using cached app...`);
+      return cachedAppPath;
+    }
+
+    if (!build.artifacts?.applicationArchiveUrl) {
+      throw new Error('Build does not have an application archive url');
+    }
+
+    return await downloadAndMaybeExtractAppAsync(
+      build.artifacts.applicationArchiveUrl,
+      platform,
+      cachedAppPath
+    );
+  }
+}
+
+async function resolvePlatformAsync(
+  nonInteractive: boolean,
+  platform?: string
+): Promise<AppPlatform> {
+  if (nonInteractive && !platform) {
+    throw new Error('Platform must be provided in non-interactive mode');
+  }
+
+  if (platform && Object.values(AppPlatform).includes(platform.toUpperCase() as AppPlatform)) {
+    return platform.toUpperCase() as AppPlatform;
+  }
```

Comment:
nit: a

```ts
const PlatformToAppPlatform: Record<Platform, AppPlatform> = {
  [Platform.IOS]: AppPlatform.Ios,
  [Platform.ANDROID]: AppPlatform.Android,
};
```

feels cleaner / safer.

EDIT: Actually, there's s `toAppPlatform` already.

Reply:
Updated to use `toAppPlatform `